components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQConnectionConsumer.java

```diff
@@ -412,5 +390,20 @@ private void handleException(String msg, Exception e) {
         log.error(msg, e);
         throw new RabbitMQException(msg, e);
     }
+}
+
+enum RabbitMQAckStates {
+    ACK,
+    REJECT,
+    REJECT_AND_REQUEUE
+}
 
-}
\ No newline at end of file
+class AckStatesAndMediationError {
```

Comment:
Would just State be a better name? It'll have the fields, state.acknowledge and state.mediationError. WDYT?

Reply:
Done in d926aaf94be8b06f295411ec90556f84d278cf52